### DISCUSSION TITLE
WECONNECT_* attributes in myconfig.env mandatory when retrieving data from WeConnect-MQTT?

### DISCUSSION THREAD

**Discussion by eili77** [tillsteinbach/VWsFriend#710] [Q&A] [answered]
Hi,
are the WECONNECT_* attributes in myconfig.env still mandatory, if I use the option in VWsFriend to retrieve the data from WeConnect-MQTT?

**Comment by tillsteinbach** [comment]
That is a misunderstanding. VWsFriend is not fetching the data from WeConnect-MQTT, instead you can enable MQTT inside VWsFriend to not have two simultaneous connections to the VW Server

**Reply by eili77** [reply]
Thanks for you swift reply and just a follow-up question to make sure I understand it correctly how data is flowing (and sorry if these maybe basic questions ;-) ):

1. WeConnect-MQTT instance would continuously pull/receive data from the VW servers
2. The received data flows from WeConnect-MQTT to my MQ-Broker
3. VWsFriend (and other subscribed consumers) would get the data from my MQ-Broker 

Is this about right?

**Reply by tillsteinbach** [reply]
No you don't need weconnect-MQTT anymore. VWsFriend is sending the data to your broker.

**Reply by eili77** [reply]
would the following be the right way to define the additional parameters in .env file for VWsFriend:?
ADDITIONAL_PARAMETERS=--mqttbroker **_ip-address_** --mqttport **1883** --mqtt-username **_user_** --mqtt-password **_pwd_** --prefix **weconnect**

**Reply by tillsteinbach** [reply]
yes, did not check in detail but looks good. Basically you can copy the parameters you use now in weconnect-mqtt

**Reply by eili77** [reply]
I used exactly the same but it doesn't work.
I re-checked and used the parameters again in weconnect-mqtt and there they work.
I also looked into the (docker) logs and I couldn't find any mqtt related records eventhough I set the log_level to debug.
Any further tip what could be checked/done?
Many thanks in advance!

**Reply by eili77** [reply, marked as answer]
ok, found the problem - well it was because of my inexperience of handling docker containers.
I thought to active the changed .env file, I need to stop and start the container... when I used "compose down" and "compose up" instead - everything works.
Apologies for the confusion and thanks for the patience - lessen learnt ;-)
